options/docker.go

```diff
@@ -27,14 +30,17 @@ func (opts *Docker) Validate() error {
 	catcher := grip.NewBasicCatcher()
 	catcher.NewWhen(opts.Port < 0, "port must be positive value")
 	catcher.NewWhen(opts.Image == "", "Docker image must be specified")
-	if opts.Platform == "" {
-		if PlatformSupportsDocker(runtime.GOOS) {
-			opts.Platform = runtime.GOOS
+	if opts.OS == "" {
+		if OSSupportsDocker(runtime.GOOS) {
+			opts.OS = runtime.GOOS
 		} else {
-			catcher.Errorf("failed to set default platform to current runtime platform '%s' because it is unsupported", opts.Platform)
+			catcher.Errorf("cannot set default OS to current runtime OS '%s' because it is unsupported", opts.OS)
 		}
-	} else if !PlatformSupportsDocker(opts.Platform) {
-		catcher.Errorf("unrecognized platform '%s'", opts.Platform)
+	} else if !OSSupportsDocker(opts.OS) {
+		catcher.Errorf("unrecognized OS '%s'", opts.OS)
+	}
+	if opts.Arch == "" {
+		opts.Arch = runtime.GOARCH
```

Comment:
Should the arch only be set to the runtime arch only if we set the OS to the runtime OS?

Reply:
I thought about this and while it would be kind of strange for a user to set OS xor arch, I think understanding the default values is simpler if they remain independent. I don't feel that strongly about it either way.